### PR TITLE
Adding entity:propIgnite()

### PR DESCRIPTION
**Pull request by Claydo9**

### CODE REVIEW

File: lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua [hunk 1]
```diff
@@ -52,4 +52,18 @@ e2function void entity:ejectPodTo( vector pos )
 
     driver:ExitVehicle()
     driver:SetPos( clampedPos )
+end
+
+-- Prop Entities ?
```
Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua]: this comment probably isnt needed

File: lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua [hunk 2]
```diff
@@ -52,4 +52,18 @@ e2function void entity:ejectPodTo( vector pos )
...
+e2function void entity:propIgnite( number duration )
+	local minDuration = 1
+	local maxDuration = 500
+
+	duration = math.Clamp(duration, minDuration, maxDuration)
```
Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua]:
```suggestion
	duration = math.Clamp( duration, minDuration, maxDuration )
```

File: lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua [hunk 3]
```diff
@@ -52,4 +52,18 @@ e2function void entity:ejectPodTo( vector pos )
...
+    
+    if not IsValid( this ) then return end
+    if this:CPPIGetOwner() ~= self.player then return end
+
+    this:Ignite(duration)
```
Comment (PLally) [inline lua/entities/gmod_wire_expression2/core/custom/cfc_entity.lua]:
```suggestion
    this:Ignite( duration )
```

**Comment by brandonsturgeon** [comment]
Closing this in favor of Decline's PR from earlier. Thank you for taking the time to write this up! I'm sorry we weren't able to use it!

**Pull request closed without merging**